test/lib/sdkdriver/test/unit/spec/transport/websocket/client.ts

```diff
@@ -0,0 +1,157 @@
+import WSClientTransport from '../../../../../src/transport/websocket/client';
+import { WebSocket } from '../../../mocks';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+
+describe('WSClientTransport', () => {
+  describe('#close', () => {
+    let closeEventPromise: Promise<void>;
+    let transport: WSClientTransport;
+    let wsClient: WebSocket;
+
+    before(async () => {
```

Comment:
Should we `beforeEach` here?

Reply:
@markandrus The tests specified within the `it()` functions here are independent enough of each other that I don't think we need to re-run this for each of them.